src/gluonts/mx/trainer/callback.py

```diff
@@ -338,3 +353,108 @@ def on_network_initializing_end(
         self, training_network: nn.HybridBlock
     ) -> None:
         copy_parameters(self.predictor.prediction_net, training_network)
+
+
+class TrainingTimeLimit(Callback):
+    def __init__(
+        self,
+        time_limit: int,
+        track_validation_duration: bool = True,
+        stop_during_epoch: bool = False,
+    ) -> None:
+        """
+        Used when you want to set a time limit to the training process.
+        Once passed into model.train(), the training process will end roughly after 'time_limit' seconds.
+
+        Attributes
+        ----------
+        time_limit: int
+            time in seconds, after which training ends
+        track_validation_duration: bool, default = True
+            whether we are recording the time used for validation.
+        stop_during_epoch: bool, default = False
+            whether we want to stop in the middle of an epoch when time is used up.
+        """
+        self.time_limit = time_limit
+        self.track_validation_duration = track_validation_duration
+        self.stop_during_epoch = stop_during_epoch
+        self.time_spent = 0.0
+        self.checkpoint = -1.0
+        self.report_exceed_timelimit = True
+
+    def on_train_start(self, max_epochs: int) -> None:
+        self.checkpoint = time.time()
+
+    def should_continue_by_timelimit(self, record_time=True, should_stop=True):
```

Comment:
I think this should be two separate methods, which do not have default arguments. It's not really clear to me at a quick glance what this is supposed to be doing.